src/QirRuntime/lib/Tracer/README.md

```diff
@@ -0,0 +1,177 @@
+# Resource Tracer Design Document #
+
+The purpose of the Resource Tracer is to provide efficient and flexible way to estimate resources of a quantum program
+ in QIR representation. The estimates are calculated by simulating execution of the program (as opposed to the static
+ analysis).

```

Comment:
The first-most reading would be easier if more details are added to the first paragraph. In particular:
> The purpose ... is ... to **estimate resources** of a ... program ...

I (as a first-time reader) would be thankful if you could provide more details about why we need this estimation (the description of the currently existing problem that implies that we need the estimation).
You could also provide the examples of the resources we plan to estimate (e.g. how much memory we use, how many qubits we allocate, how large stack is required, how many times certain functions are called). And for each item a few words about which problem this particular estimation solves. Such an info would prepare me for the expectations from the tracer (in the text below) and be ready to understand the implementation details of the tracer.

Reply:
Added link to https://docs.microsoft.com/en-us/azure/quantum/user-guide/machines/resources-estimator to provide more context/background.